### PR TITLE
Chatbot example for news api using llama index

### PR DESCRIPTION
**Pull request by asna-georgian**
This chatbot gets news and recent developments on a given topic. It's also possible to personalize the news or development on specific threads, like economics, M&A, etc. using a custom prompt template and the past messages (memory of the chat).

### CODE REVIEW

File: you_news_and_llama_index.ipynb [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from llama_index.core.memory import ChatMemoryBuffer\n",
+    "from llama_index.core.chat_engine import CondensePlusContextChatEngine\n",
+    "from llama_index.llms.openai import OpenAI\n",
+    "\n",
+    "memory = ChatMemoryBuffer.from_defaults(token_limit=3900)\n",
+    "llm = OpenAI(model=\"gpt-3.5-turbo\")\n",
```
Comment (rodrigo-georgian) [inline you_news_and_llama_index.ipynb:268]: Nit: make the output as good as possible! It's not expensive.

```suggestion
    "llm = OpenAI(model=\"gpt-4o\")\n",
```

File: you_news_and_llama_index.ipynb [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+    "You're a top-tier reporter focused on news about {question}.\\n\n",
+    "YOU SHOULD ALWAYS RESEARCH THE FOLLOWING QUERIES: \\n\n",
+    "1. {question} recent news and latest development\\n\n",
+    "2. recent regulatory changes affecting {question}\\n\n",
+    "3. recent economic events impacting {question}\\n\n",
+    "\"\"\"\n",
+    ")\n",
+    "\n",
+    "memory = ChatMemoryBuffer.from_defaults(token_limit=3900)\n",
+    "llm = OpenAI(model=\"gpt-3.5-turbo\")\n",
```
Comment (rodrigo-georgian) [inline you_news_and_llama_index.ipynb:438]: Same:

```suggestion
    "llm = OpenAI(model=\"gpt-4o\")\n",
```

**Review by rodrigo-georgian — APPROVED** [review]
Looks good to me! Left a minor suggestion to change the model to 4o but apart from that I think it's good to go. Thanks Asna!

File: you_news_and_llama_index.ipynb (hunk 2 above)
Comment (asna-georgian) [reply you_news_and_llama_index.ipynb]: done!

**Comment by asna-georgian** [comment]
Updated code to reflect YouRetriever [here](https://github.com/run-llama/llama_index/pull/13934#issuecomment-2152776960).

**Pull request merged**